### ISSUE TITLE
Evaluate `key` attributes on all react components

### ISSUE THREAD

**Issue by pdehaan** [mozilla/activity-streams#91]
I was just devTool's-in around trying to determine why there was no hover-state on the Spotlight item's title or a clickable image, when I noticed that we're using `key={site.url}` for `<SpotlightItem>` and therefore our reactid's are absurdly long.

We may want to reel that back slightly and use a numerical index instead.
I traced it back to [/content-src/components/Spotlight/Spotlight.js:37](https://github.com/mozilla/activity-streams/blob/9e9e50ac099d1e1437ded431f8f8a7bd63fad277/content-src/components/Spotlight/Spotlight.js#L37).

Changing it from this:
```js
{props.sites.map(site => <SpotlightItem key={site.url} {...site} />)}
```

... to this:
```js
{props.sites.map((site, idx) => <SpotlightItem key={idx} {...site} />)}
```

Seems to help out greatly. This is a super-nit, but I know that if I don't file it now while I see it, I'll probably forget forever. 

---

![nightly_and_activity-streams_spotlight_js_at_1d8ad808ca9a561f4bd44de4f6a202f26e04a56b_ _mozilla_activity-streams_and_activity-streams_routes_js_at_9e9e50ac099d1e1437ded431f8f8a7bd63fad277_ _mozilla_activity-streams_and_theintern_intern__a_n](https://cloud.githubusercontent.com/assets/557895/12967009/b6913a04-d017-11e5-85c8-1d2344cdd3ae.png)
**Figure 1:** Before: `data-reactid=".0.1.0.0.1.1.$http=2//www=1cnn=1com/videos/tv/2016/02/09/man-throws-alligator-in-wendys-wptv-dnt=1cnn"`

---

![nightly_and_spotlight_js_ _activity-streams_and_activity-streams_spotlight_js_at_9e9e50ac099d1e1437ded431f8f8a7bd63fad277_ _mozilla_activity-streams](https://cloud.githubusercontent.com/assets/557895/12967047/08413ad4-d018-11e5-9435-1afea36307e1.png)
**Figure 2:** After: `data-reactid=".0.1.0.0.1.1.$0"`

**Comment by pdehaan** [comment]
Looks like there are a few other places this may be the case as well:
https://github.com/mozilla/activity-streams/search?utf8=%E2%9C%93&q=key%3D&type=Code

Searching for `key={` on that page returns 4 results.

**Comment by pdehaan** [comment]
I attempted a fix for this in PR #92, but it used the `map()` function's `index` attribute, which I don't think is unique enough. Not sure if we should have Spotlight results return some sort of short GUID or UUID from the database. Either that, or we could possibly MD5 the URL to make it shorter. Although now it really just sounds like some micro-optimization.

**Comment by k88hudson** [comment]
I think eventually we'll get an ID for each result, which might be a good option. Let's revisit this when we get the firefox api further along

**Comment by pdehaan** [comment]
Fixed by #485
